src/python/pants/jvm/target_types.py

```diff
@@ -426,6 +436,135 @@ def validate(self) -> None:
+
+
+class PomXmlSourceField(SingleSourceField):
+    default = "pom.xml"
+    required = False
+
+
+class JvmArtifactsPackageMappingField(DictStringToStringSequenceField):
+    alias = "package_mapping"
+    help = help_text(
+        f"""
+        A mapping of jvm artifacts to a list of the packages they provide.
+
+        For example, `{{"com.google.guava:guava": ["com.google.common.**"]}}`.
+
+        Any unspecified jvm artifacts will use a default. See the
+        `{JvmArtifactPackagesField.alias}` field from the `{JvmArtifactTarget.alias}`
+        target for more information.
+        """
+    )
+    value: FrozenDict[str, tuple[str, ...]]
+    default: ClassVar[Optional[FrozenDict[str, tuple[str, ...]]]] = FrozenDict()
+
+    @classmethod
+    def compute_value(  # type: ignore[override]
+        cls, raw_value: dict[str, Iterable[str]], address: Address
+    ) -> FrozenDict[tuple[str, str], tuple[str, ...]]:
+        value_or_default = super().compute_value(raw_value, address)
+        assert value_or_default is not None
+        return FrozenDict(
+            {
+                cls._parse_coord(coord): tuple(packages)
+                for coord, packages in value_or_default.items()
+            }
+        )
+
+    @classmethod
+    def _parse_coord(cls, coord: str) -> tuple[str, str]:
+        group, artifact = coord.split(":")
+        return group, artifact
+
+
+class JvmArtifactsTargetGenerator(TargetGenerator):
+    alias = "jvm_artifacts"
+    core_fields = (
+        PomXmlSourceField,
+        JvmArtifactsPackageMappingField,
+        *COMMON_TARGET_FIELDS,
+    )
+    generated_target_cls = JvmArtifactTarget
+    copied_fields = COMMON_TARGET_FIELDS
+    moved_fields = (JvmArtifactResolveField,)
+    help = help_text(
+        """
+        Generate a `jvm_artifact` target for each dependency in pom.xml file.
+        """
+    )
+
+
+class GenerateFromPomXmlRequest(GenerateTargetsRequest):
+    generate_from = JvmArtifactsTargetGenerator
+
+
+@rule(
+    desc=("Generate `jvm_artifact` targets from pom.xml"),
+    level=LogLevel.DEBUG,
+)
+async def generate_from_pom_xml(
+    request: GenerateFromPomXmlRequest,
+    union_membership: UnionMembership,
+) -> GeneratedTargets:
+    generator = request.generator
+    pom_xml = await Get(
+        SourceFiles,
+        SourceFilesRequest([generator[PomXmlSourceField]]),
+    )
+    files = await Get(DigestContents, Digest, pom_xml.snapshot.digest)
+    mapping = request.generator[JvmArtifactsPackageMappingField].value
+    coordinates = parse_pom_xml(files[0].content)
+    targets = (
+        JvmArtifactTarget(
+            unhydrated_values={
+                "group": coord.group,
+                "artifact": coord.artifact,
+                "version": coord.version,
+                "packages": mapping.get((coord.group, coord.artifact)),
+                **request.template,
+            },
+            address=request.template_address.create_generated(coord.artifact),
```

Comment:
Artifact might introduce collisions, but including group produces long target names, I'm not sure which one to use

Reply:
Looks like scala_artifact only the artifact name, so it's keep it the same here
https://github.com/pantsbuild/pants/blob/release_2.20.0.dev7/src/python/pants/backend/scala/target_types.py#L588